devTools/phpstan-baseline.neon

```diff
@@ -120,6 +120,12 @@ parameters:
 			count: 1
 			path: ../src/PhpParser/Visitor/ReflectionVisitor.php
 
+		-
+			message: '#^Parameter &\$bucket by\-ref type of method Infection\\Process\\Runner\\ParallelProcessRunner\:\:fillBucketOnce\(\) expects array\<Infection\\Process\\MutantProcessContainer\>, array\<Infection\\Process\\MutantProcessContainer\|null\> given\.$#'
```

Comment:
not sure why PHPStan doesn't understand that the array is not empty..

Reply:
I believe that's because `array_shift` may return null on line https://github.com/infection/infection/pull/2097/files#diff-b6afb6bb710fec614bf0d83522b6c12084149181b26b2551de2713e3d74019d6R224

Have you tried to explicitly defined `@param-out` ? 

FYI: https://phpstan.org/blog/enhancements-in-handling-parameters-passed-by-reference#enforcing-assigned-type

Reply:
minimal reproduced in https://phpstan.org/r/f41f03a5-778a-41fb-aa65-325d2aaa906e

and created a upstream issue: https://github.com/phpstan/phpstan/issues/13093

Reply:
will be fixed in the upcoming phpstan release with https://github.com/phpstan/phpstan-src/pull/4036